From 7911b57eae46fbcb0e0a9a6a19fd9f07f5120e76 Mon Sep 17 00:00:00 2001
From: Peter Dinda <pdinda@northwestern.edu>
Date: Wed, 23 Aug 2017 15:32:46 -0500
Subject: [PATCH 2/3] Expose internal instrumentation code via the shell

You can type "profile" at any time to see the current stats
---
 src/nautilus/shell.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/nautilus/shell.c b/src/nautilus/shell.c
index f9d2db2..55e0f9b 100644
--- a/src/nautilus/shell.c
+++ b/src/nautilus/shell.c
@@ -63,6 +63,11 @@
 #include <nautilus/gdb-stub.h>
 #endif
 
+#ifdef NAUT_CONFIG_PROFILE
+#include <nautilus/instrument.h>
+#endif
+
+
 #define MAX_CMD 80
 
 struct burner_args {
@@ -818,6 +823,10 @@ static int handle_cmd(char *buf, int n)
   if (!strncasecmp(buf,"help",4)) { 
     nk_vc_printf("help\nexit\nvcs\ncores [n]\ntime [n]\nthreads [n]\n");
     nk_vc_printf("devs | fses | ofs | cat [path]\n");
+#ifdef NAUT_CONFIG_PROFILE
+    nk_vc_printf("profile\n");
+#endif
+
     nk_vc_printf("pci list | pci raw/dev bus slot func | pci dev\n");
     nk_vc_printf("shell name\n");
     nk_vc_printf("regs [t]\npeek [bwdq] x | mem x n [s] | poke [bwdq] x y\nin [bwd] addr | out [bwd] addr data\nrdmsr x [n] | wrmsr x y\ncpuid f [n] | cpuidsub f s\n");
@@ -844,6 +853,13 @@ static int handle_cmd(char *buf, int n)
     return 0;
   }
 
+#ifdef NAUT_CONFIG_PROFILE
+  if (!strncasecmp(buf,"profile",7)) {
+    nk_instrument_query();
+    return 0;
+  }
+#endif
+
   if (!strncasecmp(buf,"vcs",3)) {
     nk_switch_to_vc_list();
     return 0;
-- 
1.9.1

